atom/browser/atom_browser_client.cc

```diff
@@ -90,14 +98,22 @@ bool AtomBrowserClient::ShouldCreateNewSiteInstance(
     content::BrowserContext* browser_context,
     content::SiteInstance* current_instance,
     const GURL& url) {
-
+  if (g_suppress_renderer_process_restart) {
```

Comment:
I wonder if it is possible to have a race condition on this variable. Imagine the following chain of events:

- renderer with `nativeWindowOpen` invokes `window.open`
- soon after, another renderer without `nativeWindowOpen` invokes `window.open`

Can you guarantee that `ShouldCreateNewSiteInstance` will be invoked for the first navigation before the second navigation?

Also, imagine two renderers with `nativeWindowOpen`  invoking `window.open` in parallel, both would cause `g_suppress_renderer_process_restart` to be set, but it seems possible that only one would read it as `true`.